Changelog 136 (#100)
intellij-rust/intellij-rust.github.io: _posts/2020-11-30-changelog-136.markdown

---
layout: post
title: "IntelliJ Rust Changelog #136"
date: 2020-11-30 10:00:00 +0300
---

## New Features

<!-- https://github.com/intellij-rust/intellij-rust/pull/6378 -->
* Introduce new experimental engine for name resolution (disabled by default).

>   The new algorithm (aka name resolution 2.0) uses the fixed-point iteration approach and aims at better support for

Reviewer: ```suggestion
  The new algorithm (aka name resolution 2.0) uses the fixed-point iteration approach and aims to improve our support for
```

  complex glob imports (`use foo::*`) and re-exports. Also, it will allow the plugin to properly expand the up-coming

>   [macros 2.0](https://rust-lang.github.io/rfcs/1584-macros.html) properly. The new algorithm works for top-level items only, so local variables, as well as type inference

Reviewer: ```suggestion
  [macros 2.0](https://rust-lang.github.io/rfcs/1584-macros.html) properly. The new algorithm works for top-level items only, so local variables as well as type inference
```

>   functionality, won’t be affected.

Reviewer: ```suggestion
  functionality won’t be affected.
```

  `Settings / Preferences | Languages & Frameworks | Rust` settings.
  The feature is in the early stages of development, so we encourage you to try
  it out and share if you face any issues. Please use `Help | Create New Issue` action in your IDE or our
  [issue tracker](https://github.com/intellij-rust/intellij-rust/issues/new) directly.

<!-- https://github.com/intellij-rust/intellij-rust/pull/6276 -->
* Support `Go to Declaration` (<kbd>Ctrl</kbd> + <kbd>B</kbd>) for Cargo [features](https://doc.rust-lang.org/cargo/reference/features.html) in
  [`cfg`](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-attribute) and
  [`cfg_attr`](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg_attr-attribute) attributes.

  {% include gif-img.html path="/assets/posts/changelog-136/6276" w="700px" %}

<!-- https://github.com/intellij-rust/intellij-rust/pull/6396 -->
* Take into account macro invocations in type context, e.g. `const FOO: foobar!() = 1`

* Compiler errors detection improvements:
  <!-- https://github.com/intellij-rust/intellij-rust/pull/6352 -->
  <!-- https://github.com/intellij-rust/intellij-rust/pull/6413 -->

>     * Detect attributes that require parentheses but have no parentheses, and suggest `Add parentheses`

Reviewer 2: ```suggestion
    * Detect attributes that require parentheses but haven't them, and suggest `Add parentheses`
```
